sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -97,7 +97,15 @@ object JDBCRDD extends Logging {
    * Returns None for an unhandled filter.
    */
   def compileFilter(f: Filter, dialect: JdbcDialect): Option[String] = {
-    def quote(colName: String): String = dialect.quoteIdentifier(colName)
+    def isEnclosedInBackticks(colName: String): Boolean =
+      colName.startsWith("`") && colName.endsWith("`")
```

Comment:
I checked `V2ScanRelationPushDown`. For v2 sources, Spark always quote the column name if it contains special chars for filter pushdown. So I think here we can just invoke the sql parser to parse it
```
val nameParts = SparkSession.active.sessionState.sqlParser.parseMultipartIdentifier(colName)
assert(nameParts.length == 1) // or throw a user-facing exception if nested column can reach here
dialect.quoteIdentifier(nameParts.head)
```

Reply:
Great! I've changed it